src/pages/storage/StorageVolumeForm.tsx

```diff
@@ -0,0 +1,298 @@
+          notify.failure("Storage volume creation failed", e);
+        });
+    },
+  });
+
+  const submitForm = () => {
+    void formik.submitForm();
+  };
+  const getFormProps = (id: keyof StorageVolumeFormValues) => {
+    return {
+      id: id,
+      name: id,
+      onBlur: formik.handleBlur,
+      onChange: formik.handleChange,
+      value: formik.values[id] ?? "",
+      error: formik.touched[id] ? (formik.errors[id] as ReactNode) : null,
+      stacked: true,
+    };
+  };
+
+  const poolDriver =
+    storagePools.find((item) => item.name === pool)?.driver ?? "";
+
+  return (
+    <BaseLayout title="Create volume">
+      <NotificationRow />
+      <Row>
+        <Form onSubmit={formik.handleSubmit} stacked>
+          <Input {...getFormProps("name")} type="text" label="Name" required />
+          <StoragePoolSelector
+            value={formik.values.pool}
+            setValue={(val) => void formik.setFieldValue("pool", val)}
+            project={project}
+            selectProps={{
+              stacked: true,
+              label: "Storage pool",
+            }}
+          />
+          <Input
+            {...getFormProps("size")}
+            type="number"
+            help="Size of the storage volume"
+            label="Size in GiB"
+          />
+          <Input
+            {...getFormProps("security_shifted")}
+            type="checkbox"
+            label="Security shifted"
+            help="Enable id shifting overlay (allows attach by multiple isolated instances)"
+            stacked
+          />
+          <Input
+            {...getFormProps("security_unmapped")}
+            type="checkbox"
+            label="Security unmapped"
+            help="Disable id mapping for the volume"
+            stacked
+          />
+          <Input
+            {...getFormProps("snapshots_expiry")}
+            type="text"
+            label="Snapshots expiry"
+            help="Controls when snapshots are to be deleted (expects expression like 1M 2H 3d 4w 5m 6y)"
+          />
+          <Input
+            {...getFormProps("snapshots_pattern")}
+            type="text"
+            label="Snapshots pattern"
+            help="Pongo2 template string which represents the snapshot name (used for scheduled snapshots and unnamed snapshots)"
+          />
+          <Input
+            {...getFormProps("snapshots_schedule")}
+            type="text"
+            label="Snapshots schedule"
+            help="Cron expression (<minute> <hour> <dom> <month> <dow>), or a comma separated list of schedule aliases <@hourly> <@daily> <@midnight> <@weekly> <@monthly> <@annually> <@yearly>"
+          />
+          {["zfs", "lvm", "ceph"].includes(poolDriver) && (
+            <>
+              <Input
+                {...getFormProps("block_filesystem")}
+                type="text"
+                label="Block filesystem"
+                help="Filesystem of the storage volume"
+              />
```

Comment:
After getting the following error, I think a `Select` would be a better choice here.

![image](https://github.com/canonical/lxd-ui/assets/56583786/344cbe19-d171-4c42-bb69-dfc2205258ba)